### PR TITLE
PCIE-108-Implement-normalFlowReciever-function

### PR DESCRIPTION
**Pull request by MMostafa-Hub**

### CODE REVIEW

File: PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/layers/thread_functions.cpp [hunk 1]
```diff
@@ -5,4 +5,11 @@ void normalFlowSender(Transmitter& transmitter, QueueWrapper<TLP*>& listenOn) {
 		// popAll() waits if the queue is empty
 		transmitter.sendSequence(listenOn.popAll());
 	}
+}
+
+
+void normalFlowReceiver(Receiver& receiver, QueueWrapper<Flit*>& listenOn) {
+	while (true) {
+		receiver.receiveFlit(*listenOn.pop());
```
Comment (MMostafa-Hub) [inline PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/layers/thread_functions.cpp:13]: `receiver.recieveFlit()` doesn't take the Flits as `std::queue<Flit*>` but it takes it as `Flit` (copy). should we change it?

Comment (mohanadismail) [reply PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/layers/thread_functions.cpp:13]: That's right it should be changed

Comment (mohanadismail) [reply PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/layers/thread_functions.cpp:13]: I think receiveFlit is still not merged though. It's waiting on receivePayloadFlit

Comment (MMostafa-Hub) [reply PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/layers/thread_functions.cpp:13]: > I think receiveFlit is still not merged though. It's waiting on receivePayloadFlit
It's merged, It's only a signature without an implementation 😢

Comment (mohanadismail) [reply PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/layers/thread_functions.cpp:13]: > Should be changed to `std::queue<Flit*>` or just a `Flit*`
> 
> > That's right it should be changed

std::queue<Flit*>

**Pull request closed without merging**